lib/connectors_async/secret_storage.rb

```diff
@@ -0,0 +1,21 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+module ConnectorsAsync
+  class SecretStorage
```

Comment:
I am wondering, what will be the difference between secret storage and storage for the connector? 

Asking this because looking at the future, I am kind of wondering if we want to encrypt some values at the *storage* level, or at the *field* level.

In other words, if we were to store configuration fields (or any session data) for a connector, do we want to treat all fields through the same key/value interface and make the encryption story a separate concern?

Reply:
Sounds good! I have added a space time idea around these topics btw :)

Reply:
I think this is just in-memory storage? It will make sense to encrypt the secrets if we want to persist them, which might be true for our new ingestion service.

I'm interested to know how other integrations using Agent/Fleet/Package handle data storage(yml? how to store sensitive data?). This is part of my space time topic.